amr-wind/utilities/console_io.cpp

```diff
@@ -214,4 +215,61 @@ void print_tpls(std::ostream& out)
     }
 }
 
+void print_nonlinear_residual(
+    CFDSim& sim, ScratchField& vel_diff, ScratchField& vel_star)
+{
+    const int nlevels = sim.repo().num_active_levels();
+    const auto& mesh = sim.mesh();
+
+    const auto& velocity_new = sim.pde_manager().icns().fields().field;
+    const auto& oset_mask = sim.repo().get_int_field("mask_cell");
+
+    for (int lev = 0; lev < nlevels; ++lev) {
+
+        amrex::iMultiFab level_mask;
+        if (lev < nlevels - 1) {
+            level_mask = makeFineMask(
+                mesh.boxArray(lev), mesh.DistributionMap(lev),
+                mesh.boxArray(lev + 1), mesh.refRatio(lev), 1, 0);
+        } else {
+            level_mask.define(
+                mesh.boxArray(lev), mesh.DistributionMap(lev), 1, 0,
+                amrex::MFInfo());
+            level_mask.setVal(1);
+        }
+
+        const auto& velnew_arr = velocity_new(lev).const_arrays();
+        const auto& velstar_arr = vel_star(lev).const_arrays();
+        const auto& veldiff_arr = vel_diff(lev).arrays();
+        const auto& levelmask_arr = level_mask.const_arrays();
+        const auto& osetmask_arr = oset_mask(lev).const_arrays();
+
+        amrex::ParallelFor(
+            velocity_new(lev), amrex::IntVect(0), AMREX_SPACEDIM,
+            [=] AMREX_GPU_DEVICE(int nbx, int i, int j, int k, int n) noexcept {
+                if (osetmask_arr[nbx](i, j, k) == 0) {
+                    veldiff_arr[nbx](i, j, k, n) = 0.;
+                } else {
+                    veldiff_arr[nbx](i, j, k, n) =
+                        (velnew_arr[nbx](i, j, k, n) -
+                         velstar_arr[nbx](i, j, k, n)) *
+                        levelmask_arr[nbx](i, j, k);
+                }
+            });
+    }
+
+    amrex::Array<amrex::Real, AMREX_SPACEDIM> rms_vel;
+
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) rms_vel[idim] = 0.;
```

Comment:
remove this line. When declaring just fill it with zeros there. `rms_vel = {0.0};`

Reply:
Fixed this.